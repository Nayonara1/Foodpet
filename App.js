import React from "react";
import { createNativeStackNavigator } from '@react-navigation/native-stack'
import { NavigationContainer } from '@react-navigation/native'
import Inicial from './src/pages/Inicial'
import Principal from './src/pages/Principal'
import Cadastro from './src/pages/Cadastro'

const Stack = createNativeStackNavigator()

export default () => {
    return (
        <NavigationContainer>
            <Stack.Navigator initialRouteName='Inicial'>
                <Stack.Screen name="Inicial" component={Inicial} />
                <Stack.Screen name="Principal" component={Principal} options={{ headerBackVisible: false }}/>
                <Stack.Screen name="Cadastro" component={Cadastro}  />
            </Stack.Navigator>
        </NavigationContainer>
    )
}






















/*import React, { useState, useEffect } from "react";
import { View, Text, TouchableOpacity,StyleSheet, FlatList, Image } from "react-native";
import { FontAwesome, MaterialCommunityIcons } from "@expo/vector-icons";
import { useBackHandler } from '@react-native-community/hooks';
import { collection, getDocs } from "firebase/firestore";
import Paho from "paho-mqtt"
import Constants from 'expo-constants';


// const database = firebase.firestore();
// var clientID = "ID-" + Math.round(Math.random() * 1000);
const client = new Paho.Client(
    'broker.emqx.io',
    8083,
    '/'
)
// 'broker.emqx.io',
// 8083,
// // '10.44.1.35',
// // 9001,
// '/'
// // clientID
client.connect({
    onSuccess: function () {
        console.log("connected")
        // console.log(clientID)
        // client.subscribe("esp32/output")
        // client.subscribe("esp32/distance")
        client.subscribe("alimentador"); // As linhas a seguir sao uma tentativa de envio de mensagem
    },
    onFailure: function () {
        console.log("Desconectado")
    },
    // userName: 'emqx',
    // password: 'public',
    // useSSL: true,
})

function ligar() {
    const message1 = new Paho.Message("on"); // AGORA funcionando
    message1.destinationName = "alimentador"; // para testar

    client.send(message1); // abrir o broker online
}

function desligar() {
    const message1 = new Paho.Message("off"); // AGORA funcionando
    message1.destinationName = "alimentador"; // para testar

    client.send(message1); // abrir o broker online
}

export default function App({ navigation }) {



    return (
        
        <View style={styles.container}>
            <TouchableOpacity onPress={ligar} >
                <Text>ligar</Text>
            </TouchableOpacity>
            <TouchableOpacity onPress={desligar}>
                <Text>desligar</Text>
            </TouchableOpacity>
        </View>
    )

}


const styles = StyleSheet.create({
    container: {
      flex: 5,
      justifyContent: 'center',
      paddingTop: Constants.statusBarHeight,
      backgroundColor: 'pink',
      padding: 110,
    },
    paragraph: {
      margin: 35,
      fontSize: 18,
      fontWeight: 'center',
      textAlign: 'center',
    },
  });

*/
