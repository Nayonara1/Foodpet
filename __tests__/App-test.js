import renderer from 'react-test-renderer';
import App from '../App';
jest.mock("react-native/Libraries/Animated/NativeAnimatedHelper");
jest.useFakeTimers();
describe('Teste de Snapshot', () => {
    it('Snapshot de App', () => {
        const tela = renderer.create(<App/>).toJSON();
        expect(tela).toMatchSnapshot();
    })
})