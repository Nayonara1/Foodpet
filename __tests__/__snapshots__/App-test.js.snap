// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teste de Snapshot Snapshot de App 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="#FA6252"
        color="#fff"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Login"
        titleColor="#fff"
        titleFontSize={25}
        titleFontWeight="bold"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 250,
                "height": 250,
                "justifyContent": "center",
                "width": 250,
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "borderRadius": 12,
                  "height": 160,
                  "width": 200,
                }
              }
            />
          </View>
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
                "margin": 15,
              }
            }
          >
            Bem-vindo ao FoodPet!
          </Text>
          <TextInput
            onChangeText={[Function]}
            placeholder="  Digite seu usuário"
            style={
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 12,
                "fontSize": 16,
                "fontWeight": "bold",
                "marginBottom": 10,
                "padding": 10,
                "width": 320,
              }
            }
            testID="campoemail"
            type="text"
            value=""
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 14,
                "flexDirection": "row",
                "marginBottom": 10,
                "marginLeft": "auto",
                "marginRight": "auto",
                "width": 320,
              }
            }
          >
            <TextInput
              onChangeText={[Function]}
              placeholder="  Digite sua senha"
              secureTextEntry={true}
              style={
                Object {
                  "backgroundColor": "#fff",
                  "borderRadius": 12,
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "padding": 10,
                  "width": 250,
                }
              }
              testID="camposenha"
              type="text"
              value=""
            />
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "marginLeft": 15,
                  "opacity": 1,
                  "padding": 10,
                }
              }
            >
              <Text />
            </View>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FA6252",
                "borderRadius": 12,
                "fontSize": 18,
                "fontWeight": "bold",
                "justifyContent": "center",
                "marginBottom": 10,
                "opacity": 1,
                "padding": 10,
                "width": 320,
              }
            }
            testID="Inicial"
          >
            <Text
              style={
                Object {
                  "fontSize": 18,
                  "fontWeight": "bold",
                }
              }
            >
               Entrar 
            </Text>
          </View>
          <Text
            style={
              Object {
                "color": "#000000",
                "fontSize": 16,
                "marginTop": 20,
              }
            }
          >
            Não está cadastrado ?
            <Text
              onPress={[Function]}
              style={
                Object {
                  "color": "#0000CD",
                  "fontSize": 16,
                  "marginLeft": 10,
                }
              }
              testID="linkcadastrar"
            >
              Cadastre-se agora
            </Text>
          </Text>
          <Text
            style={
              Object {
                "color": "#000000",
                "fontSize": 16,
                "marginTop": 20,
              }
            }
          >
            Esqueceu a senha?
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#0000CD",
                    "fontSize": 16,
                    "marginLeft": 10,
                  }
                }
                testID="linkredefinir"
              >
                Redefinir senha
              </Text>
            </View>
          </Text>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
